src/Node/Expression/Test/DefinedTest.php

```diff
@@ -47,7 +47,7 @@ public function __construct(\Twig_NodeInterface $node, $name, \Twig_NodeInterfac
         } elseif ($node instanceof ConstantExpression || $node instanceof ArrayExpression) {
             $node = new ConstantExpression(true, $node->getTemplateLine());
         } else {
-            throw new SyntaxError('The "defined" test only works with simple variables.', $this->getTemplateLine(), null, null, false);
+            throw new SyntaxError('The "defined" test only works with simple variables.', $this->getTemplateLine(), $this->getSourceContext());
```

Comment:
wouldn't this break because they are not set yet here (as we are in the constructor) ? this should use `$lineno` for the line btw.

Reply:
removed for now